deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c

```diff
@@ -377,12 +391,18 @@ GrB_Info GxB_MatrixTupleIter_reset
 ) {
 	GB_WHERE1("GxB_MatrixTupleIter_reset (iter)") ;
 	GB_RETURN_IF_NULL(iter) ;
-	if(iter->iterators_size == 0) return GrB_INVALID_VALUE ;
-	iter->iterators_idx = 0;
+	if(iter->n == 0) return GrB_INVALID_VALUE ; // no iterators to iterate over is invalid scenarios
+	iter->idx = 0;
 
 	GrB_Info rv = GrB_SUCCESS ;
-	for(size_t i = 0; i < iter->iterators_size; i++)
-		rv |= GxB_MatrixTupleIter_init(&iter->iterators[i], iter->iterators[i].A) ;
+	for(size_t i = 0; i < iter->n; i++) {
+		int sparsity_type;
+		GxB_Matrix_Option_get(iter->iterators[i].A, GxB_SPARSITY_CONTROL, &sparsity_type) ;
+		if(sparsity_type != GxB_SPARSE && sparsity_type != GxB_HYPERSPARSE)
+			GB_ERROR (GrB_INVALID_VALUE, "Invalid sparsity type: %d", sparsity_type) ;
+		if((rv = _MatrixTupleIter_init(&iter->iterators[i], iter->iterators[i].A, sparsity_type)))
+			return rv;
+	}
```

Comment:
The assumption here is that we're reseting a valid iterator, as such `if(iter->n == 0)` is redundant,
also I think we save `sparsity_type` within the iterator, no need to use `GxB_Matrix_Option_get` to get it.